src/block-management/block-manager.js

```diff
@@ -267,13 +270,15 @@ export class BlockManager extends React.Component<PropsType, StateType> {
 
 	renderItem( value: { item: string, index: number } ) {
 		const clientId = value.item;
+		const testID = this.props.getBlockName( clientId ) + '-' + clientId;
```

Comment:
@pinarol Thanks for asking this... 

>  I think of UI tests as user's actions, so I think one should be able to construct an accessibility label without knowing the inner mechanisms of the app.

You're completely right here and I was hesitant to use clientId, there's a few reasons I went with it over index initially 

1. is because I thought it provided a more reliable way of identifying a specific block no matter where it went on the screen. 
2. This isn't a 'user facing' app really we're using as validation and I used that as leeway to be more flexible with what I can use in an accessibiliyId.

In your example the block that was originally block-1-heading now becomes block-2-heading and we wouldn't be 100% certain it's the same block. Thinking about it from the user actions point of view you mentioned and thinking about the scenarios we're going to build out in #745 I'm thinking maybe the indexed approach might be easier to validate the screen when there's quite a number of blocks in the post 🤔 they should come in back in the order present from the `driver.findElements` call so that was my plan for that going forward, what do you think?

Reply:
I understand the concern about differentiating one block from others. For UI testing I think it is enough to differentiate the blocks as much as users can do. A user can only tell the index of a specific block, type of it, and its contents. 

From my past experiences, I think it would be much easier to implement queries this way. We can query elements who has testIDs starting with "block-" and that'll give us the list of blocks. I didn't try with Appium btw, so your feedback is important on this. 

This way UI tests would be decoupled from inner logic. Let's say we are splitting a paragraph block. Currently the underlying mechanism keeps the first paragraph block's client id and creates a new block under it with a new client id. But what if it is updated one day to create new client id's for both blocks? This particular example is not likely to happen in the near future but I just wanted to underline the benefits of keeping UI testing code separate from inner app logic.

Reply:
Ahh I can agree with this, it' always best to not keep the UI test logic tightly coupled to the app logic and you've made some valid points for sticking to that here. It shouldn't be a difficult swap to use index vs clientID so I'll make that update.  

> From my past experiences, I think it would be much easier to implement queries this way. We can query elements who has testIDs starting with "block-" and that'll give us the list of blocks

Sure right now we're basically performing a similar query but using the XPath since appium doesn't provide a way to do this kind of querying out of the box.